### ISSUE TITLE
Installation on React native

### ISSUE THREAD

**Issue by hugohow** [bvaughn/redux-search#57]
Hello,
I'm a new developer. I tried in a naive way to install redux-search on RN but I have some problems with babel presets, that I cannot to get out ...
I think it's a basic problem but I didn't find what to do to solve my problem in a clean way.
All begin with that : 
![screen shot 2016-10-26 at 17 13 31](https://cloud.githubusercontent.com/assets/16501509/19732037/bf557386-9b9f-11e6-8fe0-a1373fbd5bbe.png)

Can somebody help me with this ? 

Thanks

**Comment by bvaughn** [comment]
Hi @hugohow. This type of question is better suited for Stack Overflow. It doesn't seem redux-search specific, and this project is only maintained by 1 person (me). Stack Overflow has a more active community to help answer questions. 😄

**Comment by spautz** [comment]
I spent most of a workday trying to get redux-search working well in a React Native app, with okay-but-not-ideal results. Since this page is one of the top search results for getting redux-search working in react-native (and since StackOverflow doesn't seem to have an answer, yet) I'm braindumping some notes here in case anybody else tries this.

**tl;dr: It's technically possible unless you want Workers, but it's a pain. redux-search itself is fine, but you have to make some manual modifications to get js-worker-search to run.**

-- needlessly gratuitous details follow -- 

(I'm using react-native@0.36.0, redux@3.6.0, and redux-search@2.3.2, which uses js-worker-search@1.1.0)

* `js-worker-search` requires `babel-plugin-typecheck`, `babel-preset-es2015`, and `babel-preset-stage-0`. Even though that project lists them as dependencies, I had to install each of these into my own app just to make things run.
* Due to some (webpack-related?) issues with the js-worker-search package, I had to manually edit its [`dist/js-worker-search.js#1398`](https://github.com/bvaughn/js-worker-search/blob/1.1.0/dist/js-worker-search.js#L1398): there's a line which references `global.crypto` -- but `global` is undefined when it runs. I didn't research deeply to find out why, but it seems to be a Webpack thing. Putting `global && global.crypto` into that conditional fixed it.
* At this point it's possible to use redux-search. This was my first time using the library, and I found the live-demo code to be a bit hard to follow (since it uses the same var names for different things in different places; "map" is used for like 5 different things) -- but [the actual docs](https://github.com/bvaughn/redux-search/tree/master/docs) were much clearer to me and everything in them seemed to work exactly as expected.
* Because React Native does not provide web workers, though, js-worker-search falls back to an inline/synchronous util for its searches: this works fine, and is pretty fast even with a lot of items, but I was really hoping to do the filtering as a background/async task. I spent some time trying to get `react-native-workers` working as a stand-in, but ran into a host of issues: string-only messages were a mild annoyance, packager-related issues with specifying a filename to the Worker were very troublesome, some other issues required working from the js-worker-search source instead of its dist, etc -- and in the end I was not able to make it work within a reasonable amount of time.

Personally, to get redux-search working with background workers, I think it might be faster to steal the search capabilities and utils from js-worker-search, build a new Worker setup that uses them but does everything via react-native-workers instead of web workers, and then make some (small) changes to redux-search to use that in place of js-worker-search itself. This is just a vague hunch, though -- it *feels* like it might work and it *feels* like it might take less time than I spent trying to bend and locally-modify the existing libraries in-place, but I did not actually try it out.
\</braindump>

**Comment by bvaughn** [comment]
Thanks for taking the time to share your findings, @spautz!

If you're ever interested in pairing on to get redux-search playing more nicely with React Native out of the box, let me know. I've written a couple of search implementations (js-worker-search, js-search) but never one that targeted native background workers. That would be a bit of an unknown area for me. But if we could get it working well enough, perhaps we could allow users to provide redux-search with their own search implementation (that meets some interface) to use instead of js-worker-search.

**Comment by jasan-s** [comment]
@bvaughn  any update to getting this lib working with react-native out of the box or easily?

**Comment by bvaughn** [comment]
Nope. Would still be interested in collaborating on that but don't really have the bandwidth to take it on myself at the moment.
